packages/hoppscotch-common/src/helpers/preRequest.ts

```diff
@@ -8,11 +8,63 @@ import {
   getGlobalVariables,
 } from "~/newstore/environments"
 import { TestResult } from "@hoppscotch/js-sandbox"
+import { getService } from "~/modules/dioc"
+import { SecretEnvironmentService } from "~/services/secret-environment.service"
 
-export const getCombinedEnvVariables = () => ({
-  global: cloneDeep(getGlobalVariables()),
-  selected: cloneDeep(getCurrentEnvironment().variables),
-})
+const secretEnvironmentService = getService(SecretEnvironmentService)
+
+const unsecretEnvironments = (
+  global: Environment["variables"],
+  selected: Environment
+) => {
+  const resolvedGlobalWithSecrets = global.map((globalVar, index) => {
+    const secretVar = secretEnvironmentService.getSecretEnvironmentVariable(
+      "Global",
+      index
+    )
+    if (secretVar) {
+      return {
+        ...globalVar,
+        secret: false,
+        value: secretVar.value,
+      }
+    }
+    return globalVar
+  })
+
+  const resolvedSelectedWithSecrets = selected.variables.map(
+    (selectedVar, index) => {
+      const secretVar = secretEnvironmentService.getSecretEnvironmentVariable(
+        selected.id,
+        index
+      )
+      if (secretVar) {
+        return {
+          ...selectedVar,
+          secret: false,
+          value: secretVar.value,
+        }
+      }
+      return selectedVar
```

Comment:
```suggestion
      return {
        ...selectedVar,
        value:
          (
            selectedVar as {
              value: string
              key: string
              secret: false
            }
          ).value ?? "",
      }
```

Similar to the above.